Third-party-libraries.md

```diff
@@ -0,0 +1,167 @@
+
+* Python 3.7
+* Google Chrome
+* git
+* Tools commonly found on Linux and MacOS systems like bash
+
+In our setup instructions, we walk developers through installing these.
+
+## Managed dependencies
+
+The rest of Oppia's development and production dependencies are managed by the Oppia code. These dependencies are installed to the following places:
+
+* `oppia_tools/` stores development tools that aren't used in production. This folder is a sibling of the repository root (`oppia/`), so if you are in the root of the repository, this folder is at `../oppia_tools/`. However, we'll use `oppia_tools/` here for simplicity.
+* `third_party/python_libs` stores our production Python dependencies.
+* `third_party/static` stores some of our production Node.js dependencies.
+* `node_modules` stores more production Node.js dependencies. On developer machines, it also stores development dependencies.
+* `third_party/` also stores some other production dependencies like protobuf files.
+
+When you run `python -m scripts.start`, a chain of scripts installs and/or upgrades dependencies as necessary:
+
+```text
+          +----------+
+          | start.py |
+          +----------+
+               |
+               | calls install_third_party_libs.main() when start.py is executed or imported
+               v
+  +-----------------------------+
+  | install_third_party_libs.py |
+  +-----------------------------+
+               |
+               | main() calls install_third_party.main()
+               v
+    +------------------------+
+    | install_third_party.py |
+    +------------------------+
+               |
+               | main() calls install_backend_python_libs.main()
+               v
++--------------------------------+
+| install_backend_python_libs.py |
++--------------------------------+
+```
+
+We'll look at each of these scripts below.
+
+### `start.py`
+
+When you start a development server, you execute `scripts/start.py`. This file does not itself install any dependencies, but it contains a call to `install_third_party_libs.main()` that will be executed whenever `start.py` is executed, or even imported.
+
+### `install_third_party_libs.py`
+
+#### Python packages
+
+Whenever `install_third_party_libs.py` is executed or imported, it installs the following Python packages:
+
+* pyyaml: Installed to `oppia_tools/`
+* future: Installed to `third_party/python_libs`
+* six: Installed to `third_party/python_libs`
+* certifi: Installed to `oppia_tools/`
+
+When `install_third_party_libs.main()` runs, it installs the following additional Python packages:
+
+* Installed "globally" using pip (in practice these are installed into your virtual environment):
+
+  * enum34
+  * protobuf
+  * grpcio
+
+* Installed to `oppia_tools/`:
+
+  * coverage
+  * pylint
+  * Pillow
+  * pylint-quotes
+  * webtest
+  * isort
+  * pycodestyle
+  * esprima
+  * PyGithub
+  * protobuf
+  * psutil
+  * pip-tools
+  * setuptools
```

Comment:
ditto, I wouldn't list these since they can be outdated pretty quickly.

Reply:
done